Classes/common/CDTFetchChanges.m

```diff
@@ -0,0 +1,118 @@
+//
+//  CDTFetchChanges.m
+//  
+//
+//  Created by Michael Rhodes on 31/03/2015.
+//
+//
+
+#import "CDTFetchChanges.h"
+
+#import "CDTDatastore.h"
+#import "CDTDocumentRevision.h"
+
+#import "TD_Database.h"
+
+@implementation CDTFetchChanges
+
+#pragma mark Initialisers
+
+- (instancetype)initWithDatastore:(CDTDatastore *)datastore
+        previousServerSequenceValue:(NSString *)previousServerSequenceValue
+{
+    self = [super init];
+    if (self) {
+        _datastore = datastore;
+        _previousServerSequenceValue = [previousServerSequenceValue copy];
+    }
+    return self;
+}
+
+#pragma mark Instance methods
+
+- (void)start
+{
+    CDTFetchChanges *weakSelf = self;
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        CDTFetchChanges *self = weakSelf;
+        if (self) {
+            [self syncStart]; 
+        }
+    });
+}
+
+- (void)syncStart
+{
+    TDChangesOptions options = {.limit = 500,
+        .contentOptions = 0,                                
+        .includeDocs = NO,  // we only need the docIDs and sequences, body is retrieved separately
+        .includeConflicts = FALSE,
+        .sortBySequence = TRUE};
+    
+    TD_RevisionList *changes;
+    SequenceNumber lastSequence = [_previousServerSequenceValue longLongValue];
+    
```

Comment:
perhaps should check to see if any of the changes blocks are set? If they are all null (obviously a programmer's error), this would just spin through the changes unnecessarily.  Maybe my suggestion is too cautious.